doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+its core dependencies inside the project using the
+`Meson dependency wrap system <https://mesonbuild.com/Wrap-dependency-system-manual.html>`__.
+This is used to pin specific versions of libraries such as `SDL2` to known-good
+versions, as well as assist with packaging and distribution on older and/or more
+esoteric systems (i.e: Emscripten).
+
+However, recompiling every single dependency Taisei needs every time you want to
+rebuild the project is not ideal. You can still use system dependencies
+installed through your OS's package manager (`apt`, `yum`, `brew`, etc) for
+development, local testing, or simply building the game for your own system.
+
+For convenience, ``meson`` will detect which packages are missing from your
+system and use its wrap dependency system to pull in what it can. The trade-off
+is that this can result in longer build times.
+
+For releases, however, you should rely *exclusively* on the ``meson`` wrap
+dependencies with the Taisei repository, for consistency and
+
+This is controlled through the ``--wrap-mode`` flag with ``meson``. (More
+on that later.)
+
+In summary:
+
+- Development = system-install dependencies are fine (`apt`, `brew`, etc)
+- Release = ``meson`` wrap dependencies only
+
+System Dependencies
+"""""""""""""""""""
+
+Linux
+'''''
+
+(The following example is specific to Ubuntu or Debian, but you can substitute
+`apt` for the package manager for your flavour of Linux.)
+
+The following will install the mandatory tools for building Taisei.
+
+.. code:: sh
+
+   apt update
+   apt install meson cmake build-essential
+
+
+The dependencies below are technically optional, as mentioned above in
+``Built-In Vs. System Dependencies``.
+
+Keep in mind that certain packages may have different names on different
+systems. Ensure you are installing the ``dev`` versions of those packages.
+
+.. code:: sh
+
+   apt install libsdl2-dev libsdl2-mixer-dev libogg-dev libopusfile-dev libpng-dev libzip-dev libx11-dev
+
+If your distribution of Linux uses Wayland as its default window server, ensure
+that Wayland deps are installed:
+
+.. code:: sh
+
+   apt install libwayland-dev
+
+macOS
+'''''
+
+On macOS, you must install the Xcode Command Line Tools to build Taisei for
+the platform, as it contains headers and tools that aren't included in the FOSS
+versions found in ``brew``.
+
+.. code:: sh
+
+   xcode-select --install
+
+There are additional command line tools that you'll need. You can acquire those
+by using `Homebrew <https://brew.sh/>`__.
+
+Follow the instructions for installing Homebrew, and then install the following
+tools:
+
+.. code:: sh
+
+   brew install meson cmake pkg-config docutils imagemagick pygments
+
+The following dependencies are technically optional, and can be pulled in at
+build-time, but they will reduce compile times during development, so it's
+recommended to install them.
+
+.. code:: sh
+
+   brew install freetype2 libzip opusfile libvorbis webp sdl2
```

Comment:
This one isn't 100% right either:
* libvorbis isn't needed
* libpng is missing (maybe installed by default?)
* libzstd is missing